picoquic/sockloop.c

```diff
@@ -949,6 +1039,11 @@ void* picoquic_packet_loop_v3(void* v_ctx)
                     send_buffer, send_buffer_size, &send_length,
                     &peer_addr, &local_addr, &if_index, &log_cid, &last_cnx,
                     send_msg_ptr);
+                if (quic != NULL && quic->quacker != NULL &&
+                        udp_quacker_base_stoc_is_none(quic->quacker) &&
+                        create_addr_key(addr_key, &peer_addr, &local_addr)) {
+                    udp_quacker_send_discovery(quic->quacker, &addr_key, 3);
```

Comment:
this should also check for if discovery ACK hasn't been received to make sure retransmit happens